project-chip/connectedhomeip
# Fix DNS-SD behavior relating to commissioning

#### Problem
* Fixes #8711  - Node to advertise non-commissionable mode in the DNS advertisement

#### Change overview
Restart Mdns at various points in the device lifecycle to reflect changes in commissioning mode
Fix CM=0/1 and AP=0/1 key/value pairs in commissionable node discovery DNS-SD response
Added shell command to linux apps to allow manual changes to commissioning mode

Usage: commissionee restartmdns <commissioningMode> <additionalPairing>

#### Testing
* Run lighting-app with shell enabled, and chip-tool to print discovery information

1) start uncommissioned lighting-app with shell enabled
2) chip-tool discover commissionables (check CM=0 and AC=1 values)
3) commission lighting-app successfully using chip-tool
4) chip-tool discover commissionables (check CM=0 and AC=0 values)
5) from lighting-app shell, turn on commissioning mode: "commissionee restartmdns true true"
6) chip-tool discover commissionables (check CM=1 and AC=1 values)
7) from lighting-app shell, turn off commissioning mode: "commissionee restartmdns false false"
8) chip-tool discover commissionables (check CM=0 and AC=0 values)
